opus/application/static_media/js/browse.js

```diff
@@ -312,29 +277,104 @@ var o_browse = {
             // we will hide page status loader from infiniteScroll if op-page-loading-status loader is spinning
             $(".op-page-loading-status > .loader").show();
             let orderBy =  $(this).data("slug");
+            let targetSlug = orderBy;
 
+            // get order of opusid when table header is clicked
+            let hash = o_hash.getHashArray();
+            let opusidOrder = (hash.order && hash.order.match(/(-?opusid)/)) ? hash.order.match(/(-?opusid)/)[0] : "opusid";
+            let isDescending = true;
             let orderIndicator = $(this).find("span:last");
+            let pillOrderIndicator = $(`.sort-contents span[data-slug="${orderBy}"] .flip-sort`);
             o_browse.reRenderData = true;
+
             if (orderIndicator.data("sort") === "sort-asc") {
                 // currently ascending, change to descending order
-                orderIndicator.data("sort", "sort-desc");
                 orderBy = '-' + orderBy;
             } else if (orderIndicator.data("sort") === "sort-desc") {
                 // currently descending, change to ascending order
-                orderIndicator.data("sort", "sort-asc");
+                isDescending = false;
                 orderBy = orderBy;
             } else {
                 // not currently ordered, change to ascending
-                orderIndicator.data("sort", "sort-asc");
+                isDescending = false;
             }
 
-            opus.prefs.order = orderBy;
+            // make sure opusid is always in order slug values
+            opus.prefs.order = orderBy.match(/opusid/) ? [orderBy] : [orderBy, opusidOrder];
+            o_browse.updateOrderIndicator(orderIndicator, pillOrderIndicator, isDescending, targetSlug);
+
             o_hash.updateHash();
             o_browse.renderSortedDataFromBeginning();
 
             return false;
         });
 
+        // browse sort order - remove sort slug
+        $(".sort-contents").on("click", "li .remove-sort", function() {
+            $(".op-page-loading-status > .loader").show();
+            let slug = $(this).parent().attr("data-slug");
+            let descending = $(this).parent().attr("data-descending");
+            o_browse.reRenderData = true;
+
+            if (descending == "true") {
+                slug = "-"+slug;
+            }
+            let slugIndex = $.inArray(slug, opus.prefs.order);
+            // The clicked-on slug should always be in the order list;
+            // The "if" is a safety precaution and the condition should always be true
+            if (slugIndex >= 0) {
+                opus.prefs.order.splice(slugIndex, 1);
+            }
+
+            // remove the sort pill right away
+            $(this).closest("li").remove();
```

Comment:
this is a little bit fuzzy, not terrible but maybe better if the li you need to remove has a class you can better find it on...?

Reply:
Update "li" to the class ".list-inline-item".